### PR TITLE
make pipe_log.rs more simple and clear

### PR DESCRIPTION
**Pull request by hicqu**
This PR trys to simplify `pipe_log.rs` and make it more clear. So that we can introduce other improvements easilier.

### CODE REVIEW

File: src/pipe_log.rs [hunk 1]
```diff
@@ -84,92 +124,76 @@ impl PipeLog {
...
-            let oflag = if current_file < manager.active_file_num {
-                // Open inactive files with readonly mode.
-                OFlag::O_RDONLY
-            } else {
-                // Open active file with readwrite mode.
-                OFlag::O_RDWR
-            };
-            let fd = fcntl::open(&path, oflag, Mode::S_IRWXU)
-                .map_err(|e| raft::StorageError::Other(e.into()))?;
+        for file_name in &logs[0..logs.len() - 1] {
```
Comment (Fullstop000) [inline src/pipe_log.rs:182]: I think `&logs` is enough?

File: src/pipe_log.rs [hunk 2]
```diff
@@ -311,40 +263,18 @@ impl PipeLog {
...
+        self.log_manager.wl().new_log_file(&self.dir)?;
+        self.write_header()?;
         Ok(())
     }
 
     pub fn write(&self, batch: &LogBatch, sync: bool, file_num: &mut u64) -> Result<usize> {
         if let Some(content) = batch.encode_to_bytes() {
             let bytes = content.len();
 
+            // TODO: `write` holds a big lock. Move the lock to log manager.
```
Comment (Fullstop000) [inline src/pipe_log.rs:275]: Maybe create an issue for this?

**Review by Fullstop000 — APPROVED** [review]
Rest LGTM

File: src/pipe_log.rs (hunk 1 above)
Comment (hicqu) [reply src/pipe_log.rs:182]: the last one file is handled after the for loop. That file needs to be opened with flag `RDWR`.

**Pull request merged**